### DISCUSSION TITLE
Possibility of IDE support (particular YAML based)

### DISCUSSION THREAD

**Discussion by noklam** [omry/omegaconf#1172] [Ideas]
What does it takes to support IDE features with LSP? What would be the approach? 

Top of my head these would be useful:
- Navigating through variable interpolation / resolver i.e. understanding the `$` operator and go to the definition.
- Go to reference, this however need some context, I am not sure should this work in an IDE. Namely it requires the LSP to understand which files are related in realistic examples (since you would likely separate config into multiple files). It would be good enough to have something works for a single file though.

**Comment by omry** [comment]
Hi @noklam, at this point - this is out of scope for OmegaConf.
Thanks for the idea.
